### PR TITLE
Computed Attributes

### PR DESCRIPTION
**Pull request by mail2bishnoi**
Hi Anton,

I have implemented the computed attribute screen. Please review and suggest improvements.

the position attribute list is not complete. I have added only three attributes. If the rest of the things are ok then I'll add the remaining attributes.

### CODE REVIEW

File: modern/src/settings/ComputedAttributePage.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+              variant="filled" />
+            <FormControl variant="filled" margin="normal" fullWidth>
+              <InputLabel>{t('sharedAttribute')}</InputLabel>
+              <Select 
+                value={item.attribute || ''} 
+                onChange={(e) => {
+                  const newValue = e.target.value;
+                  const positionAttribute = positionAttributes[newValue];
+                  setKey(newValue);
+                  if(positionAttribute && positionAttribute.type) {
```
Comment (tananaev) [inline modern/src/settings/ComputedAttributePage.js:52]: Please fix formatting. Also I would recommend extracting this into a separate method. Too much code for inline method.

**Review by tananaev — COMMENTED** [review]
Added some comments. Can you please also provide screenshots of both screens.

**Comment by mail2bishnoi** [comment]
Hi Anton,

Your suggestions included.
computed_attribute_list

computed_attribute

File: modern/src/settings/ComputedAttributesPage.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          <TableCell>{t('sharedDescription')}</TableCell>
+          <TableCell>{t('sharedAttribute')}</TableCell>
+          <TableCell>{t('sharedExpression')}</TableCell>
+          <TableCell>{t('sharedType')}</TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        {items.map((item) => (
+          <TableRow key={item.id}>
+            <TableCell className={classes.columnAction} padding="none">
```
Comment (tananaev) [inline modern/src/settings/ComputedAttributesPage.js:43]: We only allow admins to create/edit attributes. Can we hide this and add button if the user is not an admin?

**Review by tananaev — COMMENTED** [review]
One more comment. Everything else looks good.

**Comment by mail2bishnoi** [comment]
Done.

File: modern/src/settings/ComputedAttributesPage.js [hunk 3]
```diff
@@ -40,11 +42,15 @@ const ComputedAttributeView = ({ updateTimestamp, onMenuClick }) => {
       <TableBody>
         {items.map((item) => (
           <TableRow key={item.id}>
-            <TableCell className={classes.columnAction} padding="none">
-              <IconButton onClick={(event) => onMenuClick(event.currentTarget, item.id)}>
-                <MoreVertIcon />
-              </IconButton>
-            </TableCell>
+            {adminEnabled &&
+              (
```
Comment (tananaev) [inline modern/src/settings/ComputedAttributesPage.js:46]: Do we need these brackets? Same question for other places as well.

Comment (mail2bishnoi) [reply modern/src/settings/ComputedAttributesPage.js]: No. Removed.

**Comment by tananaev** [comment]
Looks good. Merged. Thanks for all the contributions.

**Pull request merged**